### PR TITLE
drivers: wifi/airoc: Remove dependency on Infineon SOC family selection

### PR DESCRIPTION
**Pull request by npal-cy**
Issue:
Kconfig WIFI_AIROC should not depend on infineon SOC family selection for the connection. See https://github.com/zephyrproject-rtos/zephyr/issues/77012 for details.

Fix:
- update modules/hal_infineon/Kconfig to be able enable USE_INFINEON_ABSTRACTION_RTOS for non Infineon SOC family.

- fix CONFIG_NET_STATISTICS_ETHERNET to CONFIG_NET_STATISTICS_WIFI in airoc_wifi.c

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

### CODE REVIEW

**Review by ifyall — APPROVED** [review]

File: drivers/wifi/infineon/airoc_wifi.c [hunk 1]
```diff
@@ -761,7 +761,7 @@ static const struct wifi_mgmt_ops airoc_wifi_mgmt = {
 	.disconnect = airoc_mgmt_disconnect,
 	.ap_enable = airoc_mgmt_ap_enable,
 	.ap_disable = airoc_mgmt_ap_disable,
-#if defined(CONFIG_NET_STATISTICS_ETHERNET)
+#if defined(CONFIG_NET_STATISTICS_WIFI)
```
Comment (jukkar) [inline drivers/wifi/infineon/airoc_wifi.c:764]: Could you split this change to different commit as the changes are not related to each other at all.

Comment (npal-cy) [reply drivers/wifi/infineon/airoc_wifi.c:764]: done

**Review by jhedberg — APPROVED** [review]

**Comment by ivanwagner** [comment]
How many approvals does a PR such as this need to be merged?

**Comment by jhedberg** [comment]
> How many approvals does a PR such as this need to be merged?

The assignee + 1.

You can follow which PRs are considered merge-ready here: https://merge-list.zephyrproject.io

The official (and detailed) definition of merge criteria can be found here: https://docs.zephyrproject.org/latest/project/project_roles.html#merge-criteria

**Review by jukkar — APPROVED** [review]
